ros/src/can_proc/scripts/thrust_proc.py

```diff
@@ -3,28 +3,41 @@
 from shared_msgs.msg import can_msg, thrust_control_msg, esc_single_msg
 
 def message_received(msg):
-  # This runs on a seperate thread from the pub
-  pass
-
-if __name__ == "__main__":
-  rospy.init_node('thrust_proc')
-  
   # Publish to the CAN hardware transmitter
   can_pub = rospy.Publisher('can_tx', can_msg,
```

Comment:
Now youre creating a new publisher every message received. You only need to make one. Look up the global keyword for python to see how to use it.